### PR TITLE
Add setting to switch between SentenceTransformer and Bert models.

### LINKED ISSUE #23: Fix memory issues for Bert models
Training a model based on the `RegionEmbedding` class consumes a large amount of (GPU) memory. This does not seem to be the case when using the `RegionEmbeddingSentenceTransformer` class.

Primary suspect is [this line](https://github.com/LAHTeR/document_segmentation/blob/feature/scripts/document_segmentation/model/region_embedding.py#L95):
```python
            out = self._transformer_model(
                **text_inputs.to(self._device)
            ).last_hidden_state
            cls_tokens = out[:, 0, :]  # CLS token is first token of sequence
```

It might be sufficient to call `cls_token.detach()` to release the memory used by `out`, which is a vector of shape `<batch_size> x <dimensionality> x <max_token>`.
For typical BERT models, the latter two factors are typically `768x512`.

This is computed per region on a page, hence mostly 10-30, but can go into the hundreds, hence scaling memory consumption accordingly.

### PR DESCRIPTION
**Pull request by carschno**
Closes #23

### CODE REVIEW

**Comment by carschno** [comment]
Tried calling [`detach()`](https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html) on the CLS token, but that does not seem to resolve the issue.

**Comment by carschno** [comment]
The RegionEmbedding type is not determined automatically based on the language model name.
The high memory consumption for BERT models has not been resolved.
